src/transformState.js

```diff
@@ -5,7 +5,35 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+        break;
+      case 'removeProperties':
+        removeProperties(action, state);
+        break;
+      case 'clear':
+        clear(state);
+        break;
+      default:
+        throw new Error(`Unknown action type ${action.type}.`);
+    }
+  }
+
+  return state;
+}
+
+function removeProperties(object, keysToRemove) {
+  for (const property of object.keysToRemove) {
```

Comment:
usually we call this variable this way

```suggestion
  for (const key of object.keysToRemove) {
```